### PR TITLE
Add mistral prompt

### PR DESCRIPTION
**Pull request by aboubakar14**

### CODE REVIEW

File: src/handlers/BaseHandler.py [hunk 1]
```diff
@@ -15,6 +15,13 @@ def __init__(self):
         """
         self.configuration = ConfigurationManager().get_configuration()
 
+    @abstractmethod
+    def initialize(self):
+        """
+        This method is used to initializes everything the handler needs in order to work.
```
Comment (Zorin95670) [inline src/handlers/BaseHandler.py]: initializes -> initialize

File: src/handlers/Ollama/OllamaHandler.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
+import os
+import requests
+import json
+import re
+
+from src.models.Diagram import Diagram
+from src.handlers.BaseHandler import BaseHandler
+
+
+class OllamaHandler(BaseHandler):
```
Comment (Zorin95670) [inline src/handlers/Ollama/OllamaHandler.py:10]: missing documentation

File: src/handlers/Ollama/OllamaHandler.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+    def initialize(self):
+        """
+        This method is used to initializes everything the handler needs in order to work.
```
Comment (Zorin95670) [inline src/handlers/Ollama/OllamaHandler.py]: typo

File: src/handlers/Ollama/OllamaHandler.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+        For Ollama, this method will load all the ModelFiles defined in the configuration file.
+        """
+
+        reponses = []
+        if "modelFiles" in self.configuration["ai-models"]["ollama"]:
```
Comment (Zorin95670) [inline src/handlers/Ollama/OllamaHandler.py]: Well prefere to have a method in BaseHandler to `getModelFiles`

File: tests/unit/handlers/test_OllamaHandler.py [hunk 5]
```diff
@@ -2,7 +2,7 @@
 from unittest.mock import patch
```
Comment (Zorin95670) [inline tests/unit/handlers/test_OllamaHandler.py]: missing to move this test in `tests/unit/handlers/ollama/*`

File: src/handlers/Factory.py [hunk 6]
```diff
@@ -26,3 +26,14 @@ def get_handler(plugin_name: str):
 
         if plugin_name == "ollama":
             return OllamaHandler()
+
+    @staticmethod
+    def initializeModels():
```
Comment (Zorin95670) [inline src/handlers/Factory.py]: sonar issue to fix

File: src/handlers/Ollama/OllamaHandler.py [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+            for model_file in self.configuration["ai-models"]["ollama"]["modelFiles"]:
```
Comment (Zorin95670) [inline src/handlers/Ollama/OllamaHandler.py]: Prefer that each handlers have its own configuration, not all configuration
`self.configuration["ai-models"]["ollama"]["modelFiles"]` -> `self.configuration["modelFiles"]`

File: src/main.py [hunk 8]
```diff
@@ -12,6 +13,12 @@
 app.include_router(api_router)
 
 
+@app.on_event("startup")
+async def startup_event():
```
Comment (Zorin95670) [inline src/main.py:17]: missing documentation

File: src/handlers/BaseHandler.py (hunk 1 above)
Comment (aboubakar14) [reply src/handlers/BaseHandler.py]: done

File: src/handlers/Ollama/OllamaHandler.py (hunk 3 above)
Comment (aboubakar14) [reply src/handlers/Ollama/OllamaHandler.py]: done

File: src/handlers/Ollama/OllamaHandler.py (hunk 2 above)
Comment (aboubakar14) [reply src/handlers/Ollama/OllamaHandler.py:10]: done

File: src/handlers/Ollama/OllamaHandler.py (hunk 4 above)
Comment (aboubakar14) [reply src/handlers/Ollama/OllamaHandler.py]: As discussed, we will see how the handlers evolves before doing that

File: src/handlers/Ollama/OllamaHandler.py (hunk 7 above)
Comment (aboubakar14) [reply src/handlers/Ollama/OllamaHandler.py]: nice idea, done

File: src/main.py (hunk 8 above)
Comment (aboubakar14) [reply src/main.py:17]: done

**Review by Zorin95670 — APPROVED** [review]

**Pull request merged**